sda-commons-server-morphia/src/test/java/org/sdase/commons/server/morphia/MorphiaBundleTracingIT.java

```diff
@@ -50,12 +54,23 @@ public void shouldHaveInstrumentation() {
     assertThat(tracer.finishedSpans())
         .extracting(MockSpan::operationName)
         .contains("createIndexes", "insert");
-    assertThat(
-            tracer.finishedSpans().stream()
-                .map(s -> s.tags().keySet())
-                .flatMap(Set::stream)
-                .noneMatch(Tags.DB_STATEMENT::equals))
-        .isTrue();
+    List<Object> dbStatementTags =
+        tracer.finishedSpans().stream()
+            .map(MockSpan::tags)
+            .map(Map::entrySet)
+            .flatMap(Set::stream)
+            .filter(e -> Tags.DB_STATEMENT.getKey().equals(e.getKey()))
+            .map(Map.Entry::getValue)
+            .collect(Collectors.toList());
+    assertSoftly(
```

Comment:
This changes what the test is testing, this test is about the instrumentation, not about removing personal data. It's sufficient to check whether spans are emitted. See https://github.com/SDA-SE/sda-dropwizard-commons/blob/master/sda-commons-server-morphia/src/test/java/org/sdase/commons/server/morphia/internal/CommandSanitizerTest.java

Reply:
But I see no integration test that ensures that the sanitizer is applied to the produced spans. I think we have to split the tracing integration test into multiple test methods to cover both, the tracing itself and the absence of personal data.

My assumption from the previous assertion is, that there should be no information about the db statement in the trace. But when I corrected the type error in the assertion the test failed. So the next best guess is, that there should be no personal information in the traced db statement. Another option would be to remove the previous assertion about the db statement. But then we do no test to verify that the sanitizer is applied, do we?

Reply:
I think the condition should be something like:
```assertThat(
            tracer.finishedSpans().stream()
                .map(s -> s.tags().keySet())
                .flatMap(Set::stream)
                .anyMatch(Tags.DB_STATEMENT.getKey()::equals))
        .isTrue();```

But I think a `isNotEmpty` matcher would be better to show that we expect `DB_STATEMENT`s:

```
    assertThat(
            tracer.finishedSpans().stream()
                .map(s -> s.tags().keySet())
                .flatMap(Set::stream)
                .filter(Tags.DB_STATEMENT.getKey()::equals))
        .isNotEmpty();```